### ISSUE TITLE
HTTP 503 error when local http channel is queried after other channels

### ISSUE THREAD

**Issue by tomashek** [conda/conda#3549]
I've been receiving strange 503 errors when attempting to use simple conda commands (conda search) with multiple local channels configured. One is a "file" channel, the other a simple local http server. 

The channels, when used independently, work fine. 

The conda commands seem to hang, then fail, in the "session.get" in fetch.py's fetch_repodata. 

Observations: if I purposely order the requests such that our local http server comes first, it works fine. If ordered such that the http server request is not first, it fails. This is consistent with or without proxy, with or without cache=True, and with or without enabling concurrency in fetch_index. Also, when the operations fail, I see no indication that any requests even reach the webserver, so it seems to be a client-side failure.

It is always when the local http channel comes after some other channel (even defaults).

```
Current conda install:

               platform : linux-64
          conda version : 4.2.9
       conda is private : False
      conda-env version : 4.2.9
    conda-build version : 2.0.1
         python version : 3.5.2.final.0
       requests version : 2.9.1
       root environment : /localdisk/work/tmtomash/miniconda3  (writable)
    default environment : /localdisk/work/tmtomash/miniconda3
       envs directories : /localdisk/work/tmtomash/miniconda3/envs
          package cache : /localdisk/work/tmtomash/miniconda3/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/linux-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
                          http://fxsatmac01.fx.intel.com:5000/linux-64/
                          http://fxsatmac01.fx.intel.com:5000/noarch/
            config file : /nfs/site/home/tmtomash/.condarc
           offline mode : False
```

```
$ conda search bzip2
Fetching package metadata ...
CondaHTTPError: HTTP Error: HTTPError: 503 Server Error: Service Unavailable for url: http://fxsatmac01.fx.intel.com:5000/noarch/repodata.json: http://fxsatmac01.fx.intel.com:5000/noarch/
```

**Comment by tomashek** [comment]
I just replicated this problem when the http server was the first and only channel.  The apparent "trigger" was proxy settings in the condarc file (proxy_servers: http: https: ). If I put those same proxy configurations in the environment variables HTTP_PROXY and HTTPS_PROXY and remove them from condarc, it works. It they are in the condarc, it fails.

**Comment by tomashek** [comment]
I went back to my original reproducer and was able to resolve the problem by removing the "proxy_servers:" configuration from the condarc file. Having those same proxy settings in the environment does not show the error.

So it looks like there is a bug in the proxy implementation in conda.

**Comment by kalefranz** [comment]
This should be resolved with #3550.

**Comment by pakelley** [comment]
I've been running into intermittent 500-errors when launching Dataproc clusters with about 32 or more nodes. A few weeks ago I was running into this and it seemed to be solved by upgrading to conda 4.5, but it seems that it just made these errors less frequent.

I asked Anaconda support for help on the issue, and they sent me to this issue: I checked to see if I was setting any proxy servers in my `.condarc`, and I didn't see anything there about proxies(and I'm not intentionally setting any).

In case it helps, here's the output from `conda info -a`:

```
active environment : None
user config file : /root/.condarc
populated config files : /root/.condarc
conda version : 4.5.1
conda-build version : not installed
python version : 3.6.5.final.0
base environment : /opt/conda (writable)
channel URLs : https://conda.anaconda.org/conda-forge/linux-64
https://conda.anaconda.org/conda-forge/noarch
https://conda.anaconda.org/t/<TOKEN>/bombora/linux-64
https://conda.anaconda.org/t/<TOKEN>/bombora/noarch
https://repo.anaconda.com/pkgs/main/linux-64
https://repo.anaconda.com/pkgs/main/noarch
https://repo.anaconda.com/pkgs/free/linux-64
https://repo.anaconda.com/pkgs/free/noarch
https://repo.anaconda.com/pkgs/r/linux-64
https://repo.anaconda.com/pkgs/r/noarch
https://repo.anaconda.com/pkgs/pro/linux-64
https://repo.anaconda.com/pkgs/pro/noarch
package cache : /opt/conda/pkgs
/root/.conda/pkgs
envs directories : /opt/conda/envs
/root/.conda/envs
platform : linux-64
user-agent : conda/4.5.1 requests/2.18.4 CPython/3.6.5 Linux/3.16.0-6-amd64 debian/8 glibc/2.19
UID:GID : 0:0
netrc file : None
offline mode : False

# conda environments:
#
base * /opt/conda

sys.version: 3.6.5 |Anaconda, Inc.| (default, Mar 29 ...
sys.prefix: /opt/conda
sys.executable: /opt/conda/bin/python
conda location: /opt/conda/lib/python3.6/site-packages/conda
conda-build: None
conda-env: /opt/conda/bin/conda-env
user site dirs:

CIO_TEST: <not set>
CONDA_BIN_PATH: /opt/conda/bin
CONDA_ROOT: /opt/conda
PATH: /opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
REQUESTS_CA_BUNDLE: <not set>
SSL_CERT_FILE: <not set>
```

Am I missing anything that's likely to give me these 500 errors? And what are options, if any, for getting around this issue?
